addons/l10n_pt_stock/report/report_deliveryslip.py

```diff
@@ -0,0 +1,11 @@
+from odoo import api, models
+
+
+class ReportStockPickingDeliverySlip(models.AbstractModel):
+    _inherit = 'report.stock.report_deliveryslip'
+
+    @api.model
+    def _get_report_values(self, docids, data=None):
+        if self.env.company.account_fiscal_country_id.code == 'PT':
+            self.env['stock.picking']._l10n_pt_stock_compute_missing_hashes(self.env.company.id)
```

Comment:
This triggers the computation of ALL missing hashes, and not just the printed document.
I worry about negative impacts on concurrent warehouse operations in a busy WH.